torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -361,6 +365,10 @@ def post_backward(self, *unused: Any):
                     fsdp_params_with_grad.append(fsdp_param)
                     unsharded_grads.append(fsdp_param.unsharded_grad_data)
                     fsdp_param.unsharded_param.grad = None
+                else:
+                    continue
+                if isinstance(fsdp_param.fsdp_placement, _FlatShard):
+                    unsharded_grads[-1] = unsharded_grads[-1].view(-1)
```

Comment:
seems `.view(-1)` is the core change at fsdp side. curious should `.view(-1)` live in fsdp or dtensor? I can imagine a case where `view(-1) is done inside `DTensor.to_local()`

Reply:
the question is long-term and non-blocking. totally understnad we want to unblock in short-term

Reply:
> where view(-1) is done inside DTensor.to_local()

This unsharded grad is normally `torch.Tensor`. I think we have to do this `.view(-1)` so that the downstream post-backward logic works out of the box (e.g. computing what the padded unsharded size is for reduce-scatter). Otherwise, we may need more refactoring.